### PR TITLE
Adding RAID Parameters in api.md

### PR DESCRIPTION
**Pull request by ZainubW**
Adding RAID Parameters in api.md.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -142,6 +142,45 @@ devices.
...
+
+This field contains the information about the RAID configuration for bare 
+metal servers. 
+
+The sub-fields are
+
+* *hardwareRAIDVolumes* -- It contains the list of logical disks for hardware 
+  RAID, if rootDeviceHints isn't used, first volume is root volume. Furthermore,
+  it defines the desired configuration of volume in hardware RAID.
+  * *level* -- 'RAID level for the logical disk. The following levels are
```
Comment (dtantsur) [inline docs/api.md:155]: Redundant quotes

File: docs/api.md [hunk 2]
```diff
@@ -142,6 +142,45 @@ devices.
...
+This field contains the information about the RAID configuration for bare 
+metal servers. 
+
+The sub-fields are
+
+* *hardwareRAIDVolumes* -- It contains the list of logical disks for hardware 
+  RAID, if rootDeviceHints isn't used, first volume is root volume. Furthermore,
+  it defines the desired configuration of volume in hardware RAID.
+  * *level* -- 'RAID level for the logical disk. The following levels are 
+    supported: 0;1;2;5;6;1+0;5+0;6+0.'
```
Comment (dtantsur) [inline docs/api.md:156]: It's not clear what here is values. Maybe something like `0`, `1`, ....?

File: docs/api.md [hunk 3]
```diff
@@ -142,6 +142,45 @@ devices.
...
+  * *name* -- Name of the volume. Should be unique within the Node. If not
```
Comment (dtantsur) [inline docs/api.md:157]: The term Node has a different meaning in k8s, let's use "server" or "machine".

File: docs/api.md [hunk 4]
```diff
@@ -142,6 +142,45 @@ devices.
...
+  HardwareRAIDVolumes is set this item will be invalid. The number of created 
+  Software RAID devices must be 1 or 2. If there is only one Software RAID 
+  device, it has to be a RAID-1. If there are two, the first one has to be a 
+  RAID-1, while the RAID level for the second one can be 0, 1, or 1+0. As the 
+  first RAID device will be the deployment device, enforcing a RAID-1 reduces 
+  the risk of ending up with a non-booting node in case of a disk failure. 
+  Furthermore, SoftwareRAIDVolume defines the desired configuration of volume 
+  in software RAID.
+  * *level* -- RAID level for the logical disk. The following levels are 
+    supported: 0;1;1+0.
```
Comment (dtantsur) [inline docs/api.md:177]: See above re levels

File: docs/api.md [hunk 5]
```diff
@@ -142,6 +142,45 @@ devices.
...
+  RAID, if rootDeviceHints isn't used, first volume is root volume. Furthermore,
+  it defines the desired configuration of volume in hardware RAID.
+  * *level* -- 'RAID level for the logical disk. The following levels are 
+    supported: 0;1;2;5;6;1+0;5+0;6+0.'
+  * *name* -- Name of the volume. Should be unique within the Node. If not 
+    specified, volume name will be auto-generated.
+  * *numberOfPhysicalDisks* -- Integer, number of physical disks to use for the 
+    logical disk. Defaults to minimum number of disks required for the 
+    particular RAID level.
+  * *rotational* -- Select disks with only rotational or solid-state storage.
```
Comment (dtantsur) [inline docs/api.md:162]: Something like "If true, select only rotational disks, if false - only solid-state and NVMe. Any disk types are used by default."

**Review by dtantsur — CHANGES_REQUESTED** [review]
/ok-to-test

File: docs/api.md (hunk 2 above)
Comment (ZainubW) [reply docs/api.md:156]: ohh yes, you right. Changing it.

File: docs/api.md (hunk 3 above)
Comment (ZainubW) [reply docs/api.md:157]: okay sure.

File: docs/api.md (hunk 1 above)
Comment (ZainubW) [reply docs/api.md:155]: ohh yeah. My mistake. Thanks

File: docs/api.md [hunk 6]
```diff
@@ -142,6 +142,46 @@ devices.
...
+    supported: `0`,`1`,`2`,`5`,`6`,`1+0`,`5+0`,`6+0`.
+  * *name* -- Name of the volume. Should be unique within the server. If not 
+    specified, volume name will be auto-generated.
+  * *numberOfPhysicalDisks* -- Integer, number of physical disks to use for the 
+    logical disk. Defaults to minimum number of disks required for the 
+    particular RAID level.
+  * *rotational* -- If true, select only rotational disks, if false - only 
+    solid-state and NVMe. Any disk types are used by default.
+  * *sizeGibibytes* -- Size (Integer) of the logical disk to be created in GiB. 
+    If unspecified or set be 0, the maximum capacity of disk will be used for
```
Comment (zaneb) [inline docs/api.md:165]:
```suggestion
    If unspecified or set to 0, the maximum capacity of disk will be used for 
```

File: docs/api.md [hunk 7]
```diff
@@ -142,6 +142,46 @@ devices.
...
 
+#### raid
+
+This field contains the information about the RAID configuration for bare 
+metal servers. 
+
+The sub-fields are
+
+* *hardwareRAIDVolumes* -- It contains the list of logical disks for hardware 
+  RAID, if rootDeviceHints isn't used, first volume is root volume. Furthermore,
```
Comment (zaneb) [inline docs/api.md:153]:
```suggestion
  RAID. If rootDeviceHints isn't used, the first volume is the root volume. Furthermore,
```

File: docs/api.md [hunk 8]
```diff
@@ -142,6 +142,46 @@ devices.
...
+  * *numberOfPhysicalDisks* -- Integer, number of physical disks to use for the 
+    logical disk. Defaults to minimum number of disks required for the 
+    particular RAID level.
+  * *rotational* -- If true, select only rotational disks, if false - only 
+    solid-state and NVMe. Any disk types are used by default.
+  * *sizeGibibytes* -- Size (Integer) of the logical disk to be created in GiB. 
+    If unspecified or set be 0, the maximum capacity of disk will be used for 
+    logical disk.
+* *softwareRAIDVolumes* -- It contains the list of logical disks for software 
+  RAID, if rootDeviceHints isn't used, first volume is root volume. If
```
Comment (zaneb) [inline docs/api.md:168]:
```suggestion
  RAID. If rootDeviceHints isn't used, the first volume is the root volume. If
```

File: docs/api.md [hunk 9]
```diff
@@ -304,6 +344,7 @@ Settings related to deploying an image to the host.
 * *id* -- The unique identifier for the service in the underlying
   provisioning tool.
 * *image* -- The image most recently provisioned to the host.
+* *raid* -- The list of hardware and software RAID volumes recently set.
```
Comment (zaneb) [inline docs/api.md:347]:
```suggestion
* *raid* -- The list of hardware or software RAID volumes recently set.
```

You can't set both at the same time. We might want to mention that more prominently as well.

**Review by zaneb — COMMENTED** [review]
Thanks!
/approve

File: docs/api.md [hunk 10]
```diff
@@ -142,6 +142,46 @@ devices.
...
+  first RAID device will be the deployment device, enforcing a RAID-1 reduces 
+  the risk of ending up with a non-booting node in case of a disk failure. 
+  Furthermore, SoftwareRAIDVolume defines the desired configuration of volume 
+  in software RAID.
+  * *level* -- RAID level for the logical disk. The following levels are 
+    supported: `0`,`1`,`1+0`.
+  * *physicalDisks* -- A list of device hints, the number of items should be
+    greater than or equal to 2.
+  * *sizeGibibytes* -- Size (Integer) of the logical disk to be created in 
+    GiB. If unspecified or set be 0, the maximum capacity of disk will be
```
Comment (furkatgofurov7) [inline docs/api.md:182]:
```suggestion
    GiB. If unspecified or set to 0, the maximum capacity of disk will be 
```

File: docs/api.md [hunk 11]
```diff
@@ -142,6 +142,46 @@ devices.
 
 **NOTE:** These are subject to change.
 
+#### raid
+
+This field contains the information about the RAID configuration for bare 
+metal servers. 
+
+The sub-fields are
```
Comment (furkatgofurov7) [inline docs/api.md:150]:
```suggestion
The sub-fields are:
```

File: docs/api.md (hunk 9 above)
Comment (ZainubW) [reply docs/api.md:347]: oh yes. thanks

**Pull request merged**